apps/api/src/app/shared/services/in-memory-provider/in-memory-provider.service.ts

```diff
@@ -1,41 +1,156 @@
-import { Logger } from '@nestjs/common';
-
-import { CLIENT_READY, getRedisInstance, getRedisProviderConfig, IRedisProviderConfig, Redis } from './redis-provider';
+import { Injectable, Logger } from '@nestjs/common';
+
+import {
+  getElasticacheCluster,
+  getElasticacheClusterProviderConfig,
+  IElasticacheClusterProviderConfig,
+} from './elasticache-cluster-provider';
+import {
+  CLIENT_READY,
+  getRedisInstance,
+  getRedisProviderConfig,
+  IRedisProviderConfig,
+  Redis,
+  RedisOptions,
+} from './redis-provider';
+import {
+  Cluster,
+  ClusterOptions,
+  getRedisCluster,
+  getRedisClusterProviderConfig,
+  IRedisClusterProviderConfig,
+} from './redis-cluster-provider';
 
 import { ApiException } from '../../exceptions/api.exception';
 
 const LOG_CONTEXT = 'InMemoryCluster';
 
-export type InMemoryProviderClient = Redis | undefined;
-type InMemoryProviderConfig = IRedisProviderConfig;
+export type InMemoryProviderClient = Redis | Cluster | undefined;
+type InMemoryProviderConfig = IElasticacheClusterProviderConfig | IRedisProviderConfig | IRedisClusterProviderConfig;
 
+@Injectable()
 export class InMemoryProviderService {
   public inMemoryProviderClient: InMemoryProviderClient;
   public inMemoryProviderConfig: InMemoryProviderConfig;
 
   constructor() {
-    if (!this.inMemoryProviderClient) {
-      this.inMemoryProviderClient = process.env.IN_MEMORY_CLUSTER_MODE_ENABLED
-        ? this.inMemoryClusterProviderSetup()
-        : this.inMemoryProviderSetup();
-    }
+    this.inMemoryProviderClient = this.buildClient();
+  }
+
+  private buildClient(): Redis | Cluster | undefined {
+    return this.isClusterMode() ? this.inMemoryClusterProviderSetup() : this.inMemoryProviderSetup();
   }
 
   public isClientReady(): boolean {
     if (!this.inMemoryProviderClient) {
       return false;
     }
 
+    // TODO: Check this works for Cluster
     return this.inMemoryProviderClient.status === CLIENT_READY;
   }
 
-  private inMemoryClusterProviderSetup(): InMemoryProviderClient {
+  public isClusterMode(): boolean {
+    return process.env.IN_MEMORY_CLUSTER_MODE_ENABLED === 'true';
+  }
+
+  public getClusterOptions(): ClusterOptions | undefined {
+    if (this.inMemoryProviderClient && this.isClusterMode()) {
+      return this.inMemoryProviderClient.options;
+    }
+  }
+
+  public getOptions(): RedisOptions | undefined {
```

Comment:
Downstream the Redis Cluster can have Redis Options set up and configured just like a single instance, but they live in a different place.